### PR TITLE
Changing seed

### PR DESCRIPTION
**Pull request by CrocoNiall**
Added some ui-router resolves to ensure that conditions are met before rendering states. 

authRequired - ensures use is logged in before rendering state
currentUser - grabs the current DB record for that user, we can then inject that into any controller. 
isAdmin - only renders the view if currentUser is admin

### CODE REVIEW

**Comment by CrocoNiall** [comment]
lat is the the token that we get back from firebase. we then attach that token in a header to every HTTP request to our backend. This is how the backend knows that its an authorized user accessing data from our backend.

File: frontend/js/factories/user.factory.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ function UserFactory($http) {
   };
 }
 
-UserFactory.$inject = ['$http'];
+UserFactory.$inject = ['$http', '$rootScope'];
```
Comment (jcolairo) [inline frontend/js/factories/user.factory.js]: why have you got $rootScope here?

File: frontend/js/controllers/auth.controller.js [hunk 2]
```diff
@@ -52,9 +54,9 @@ function AuthController($state, AuthFactory, $rootScope) {
     controller.error = null;
     controller.email = '';
     controller.password = '';
-    AuthFactory.$onAuthStateChanged(function (user) {
-      controller.user = user;
-      if (user) {
+    AuthFactory.$onAuthStateChanged(function (fbUser) {
+      controller.user = fbUser;
+      if (fbUser) {
```
Comment (jcolairo) [inline frontend/js/controllers/auth.controller.js:59]: did you put the fbUser to get the information from FB

File: frontend/js/controllers/auth.controller.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-function AuthController($state, AuthFactory, $rootScope) {
+function AuthController($state, AuthFactory, $rootScope, UserFactory) {
```
Comment (jcolairo) [inline frontend/js/controllers/auth.controller.js]: Why have you got UserFactory here?

**Pull request merged**